Documentation/Diagnostics/PH2151.md

```diff
@@ -0,0 +1,101 @@
+# PH2151: Avoid TODO comments
+
+| Property | Value  |
+|--|--|
+| Package | [Philips.CodeAnalysis.MaintainabilityAnalyzers](https://www.nuget.org/packages/Philips.CodeAnalysis.MaintainabilityAnalyzers) |
+| Diagnostic ID | PH2151 |
+| Category  | [Maintainability](../Maintainability.md) |
+| Analyzer | [AvoidTodoCommentsAnalyzer](https://github.com/philips-software/roslyn-analyzers/blob/main/Philips.CodeAnalysis.MaintainabilityAnalyzers/Maintainability/AvoidTodoCommentsAnalyzer.cs)
+| CodeFix  | No |
+| Severity | Warning |
+| Enabled By Default | No |
+
+## Introduction
+
+This rule flags TODO comments in C# source code.
+
+## Reason
+
+Source code is not the place to get attention for a workitem. Use a proper task management system instead.
+
+## How to fix violations
+
+Replace TODO comments with proper task management:
+- Create tasks in your project management system (Jira, Azure DevOps, GitHub Issues, etc.)
+- Remove the TODO comment from the source code
+- If the comment contains important context, move it to the task description
+
+## Examples
+
+### Incorrect
+
+```csharp
+public class UserService
+{
+    // TODO: Implement user authentication
+    public bool AuthenticateUser(string username, string password)
+    {
+        return false;
+    }
+    
+    /* TODO: Add error handling and logging */
+    public void CreateUser(User user)
+    {
+        // implementation
+    }
+    
+    // todo: optimize this method for performance
+    public List<User> GetAllUsers()
+    {
+        // implementation
+        return new List<User>();
+    }
+}
+```
+
+### Correct
+
+```csharp
+public class UserService
+{
+    // NOTE: User authentication is tracked in issue #123
+    public bool AuthenticateUser(string username, string password)
+    {
+        return false;
+    }
+    
+    // Implementation requires error handling - see task PROJ-456
+    public void CreateUser(User user)
+    {
+        // implementation
+    }
+    
+    // Performance optimization tracked in backlog item #789
+    public List<User> GetAllUsers()
+    {
+        // implementation
+        return new List<User>();
+    }
+}
+```
+
+## Configuration
+
+This rule is disabled by default. Enable it in your `.editorconfig`:
```

Comment:
Do we follow this pattern with other md files? We shouldn't repeat this information - we already say it's disabled in the same file, above.